OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1

```diff
@@ -80,17 +84,26 @@ function Get-TentacleThumbprint
     return $thumbprint
 }
 
+function Get-TentacleVersion
+{
+    $tentaclepath = "${env:ProgramFiles}\Octopus Deploy\Tentacle"
+    # Get the thumbprint of this tentacle. original version added non-printing chars
+    $version = & $tentaclepath\tentacle.exe Version --console
+    return $version.SubString(0, 5)
```

Comment:
This seems... dangerous. What's the reasoning for just getting the first 5 chars?

Reply:
Tentacle.exe Version returns 4.0.5+Branch.master.Sha.125ef8148c1b830798af3b1fdbeff42a08d8e2de which [System.Version] wasn't able to handle.  Originally I was just checking to see if the version number was greater than 4.0.0 to determine if spaces were supported.  I have changed my approach and removed the function and check entirely and check to see if the space was provided as a parameter instead.

Reply:
In the server, we load the version number from the file - there's an "informational version" which has the branch and sha in it. But there's also a file version which has the `4.0.5`.
See https://github.com/OctopusDeploy/OctopusDSC/blob/master/OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1#L351